libsolidity/ast/AST.cpp

```diff
@@ -252,6 +252,30 @@ ContractDefinitionAnnotation& ContractDefinition::annotation() const
 	return dynamic_cast<ContractDefinitionAnnotation&>(*m_annotation);
 }
 
+std::vector<FunctionDefinition const*> ContractDefinition::definedFunctions() const
+{
+	std::vector<FunctionDefinition const*> totalFunctions = filteredNodes<FunctionDefinition>(m_subNodes);
+	std::vector<ModifierArea const*> baseModifierAreas = modifierAreas();
+	std::vector<ModifierArea const*> nextGroups;
+	while (!baseModifierAreas.empty())
```

Comment:
I think this may be easier to read, if you used a ``queue`` of ``ModifierArea const*``. Then you could extract the front element into a local variable, pop it from the queue, process it and while doing so just add the subareas back to the queue (and continue until the queue is empty).

This may only be my personal taste, though, and it should also be fine as it is now.